src/shared/dissect-image.c

```diff
@@ -2110,45 +2123,36 @@ int dissected_image_mount(
         if (r < 0)
                 return r;
 
-        xbootldr_mounted = mount_partition(PARTITION_XBOOTLDR, m->partitions + PARTITION_XBOOTLDR, where, "/boot", uid_shift, uid_range, flags);
-        if (xbootldr_mounted < 0)
-                return xbootldr_mounted;
+        int slash_boot_is_available = -ENODATA;
+        r = slash_boot_is_available = mount_point_is_available(where, "/boot");
+        if (r < 0)
+                return r;
+        if (r > 0) {
+                r = mount_partition(PARTITION_XBOOTLDR, m->partitions + PARTITION_XBOOTLDR, where, "/boot", uid_shift, uid_range, flags);
+                if (r < 0)
+                        return r;
+                slash_boot_is_available = r;
```

Comment:
hmm, shouldn't this be `!r`? after all if you just mounted XBOOTLDR to /boot then /boot is not available for anything esle.

Reply:
Arrgh. Yes.